app-editors/atom/atom-1.27.2.ebuild

```diff
@@ -0,0 +1,563 @@
+	newexe "${BUILD_DIR}/app/atom.sh" atom
+	insinto "/usr/share/licenses/${PN}${suffix}"
+	doins "${BIN_S}/$(get_atom_rpmdir)/resources/LICENSE.md"
+	dosym "${install_dir}/atom" "/usr/bin/atom${suffix}"
+	dosym "${install_dir}/app/apm/bin/apm" "/usr/bin/apm${suffix}"
+
+	_fix_executables "${install_dir}/app/apm/bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/.bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/npm/bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/npm/bin/node-gyp-bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/node-gyp/bin"
+}
+
+pkg_postinst() {
+	xdg_desktop_database_update
+}
+
+pkg_postrm() {
+	xdg_desktop_database_update
+}
+
+# Helpers
+# -------
+
+# Return the installation suffix appropriate for the slot.
+get_install_suffix() {
+	local c=(${SLOT//\// })
+	local slot=${c[0]}
+	local suffix
+
+	if [[ "${slot}" == "0" ]]; then
+		suffix=""
+	else
+		suffix="-${slot}"
+	fi
+
+	echo -n "${suffix}"
+}
+
+# Return the upstream app name appropriate for $PV.
+get_atom_appname() {
+	if [[ "${PV}" == *beta* ]]; then
+		echo -n "atom-beta"
+	else
+		echo -n "atom"
+	fi
+}
+
+# Return the app installation path inside the upstream archive.
+get_atom_rpmdir() {
+	echo -n "usr/share/$(get_atom_appname)"
+}
+
+# Return the installation target directory.
+get_install_dir() {
+	echo -n "/usr/$(get_libdir)/atom$(get_install_suffix)"
+}
+
+# Return the Electron installation directory.
+get_electron_dir() {
+	echo -n "${EROOT%/}/usr/$(get_libdir)/electron-${ELECTRON_SLOT}"
+}
+
+# Return the directory containing appropriate Node headers
+# for the required version of Electron.
+get_electron_nodedir() {
+	echo -n "${EROOT%/}/usr/include/electron-${ELECTRON_SLOT}/node/"
+}
+
+# Run JavaScript using Electron's version of Node.
+enode_electron() {
+	"$(get_electron_dir)"/node $@
+}
+
+# Run node-gyp using Electron's version of Node.
+enodegyp_atom() {
+	local apmpath="$(get_atom_rpmdir)/resources/app/apm"
+	local nodegyp="${BIN_S}/${apmpath}/node_modules/node-gyp/bin/node-gyp.js"
+
+	PATH="$(get_electron_dir):${PATH}" \
+		enode_electron "${nodegyp}" \
+			--nodedir="$(get_electron_nodedir)" $@ || die
+}
+
+# Coffee Script wrapper.
+ecoffeescript() {
+	local cscript="${FILESDIR}/transpile-coffee-script.js"
+
+	# Disable shell glob expansion, as we want the coffee script
+	# transpiler to do that instead.
+	set -f
+	echo "ecoffeescript" $@
+	ATOM_HOME="${T}/.atom" ATOM_SRC_ROOT="${BUILD_DIR}/app" \
+	NODE_PATH="${BUILD_DIR}/app/node_modules" \
+		enode_electron "${cscript}" $@ || die
+	set +f
+}
+
+# asar wrapper.
+easar() {
+	local asar="${WORKDIR}/$(package_dir asar)/node_modules/asar/bin/asar"
+	echo "asar" $@
+	enode_electron "${asar}" $@ || die
+}
+
+# Return a $WORKDIR directory for a given package name.
+package_dir() {
+	local binmod="${1//-/_}"
+	local binmod_v="${binmod^^}_V"
+	echo -n ${1}-${!binmod_v}
+}
+
+# Check if there are patches for a given package.
+_have_patches_for() {
```

Comment:
Don't start function names with `_`, this is inconsistent and makes no sense here.